src/components/converter/Form.js

```diff
@@ -31,87 +31,119 @@ const ANJ_MIN_REQUIRED = bigNum(10)
 
-  // fromInputValue and toInputValue are filtered values to be set on inputs (strings).
-  const fromInputValue = value
-  const toInputValue = formatUnits(toAmount, {
-    digits: toDecimals,
-  })
+  if (amount.lt(0)) {
+    return null
+  }
 
-  return { fromInputValue, toInputValue, fromAmount, toAmount }
+  return { amount, inputValue }
 }
 
-// Convert the two input values as the user types
-function useConvertInputs(symbol, tokenToAnjRate, anjToTokenRate) {
+// Convert the two input values as the user types.
+// The token which it is converted from is referred to as “other”.
+function useConvertInputs(otherSymbol) {
   const [inputValueAnj, setInputValueAnj] = useState('')
-  const [inputValueToken, setInputValueToken] = useState('')
+  const [inputValueOther, setInputValueOther] = useState('')
   const [amountAnj, setAmountAnj] = useState(bigNum(0))
-  const [amountToken, setAmountToken] = useState(bigNum(0))
-  const antDecimals = useTokenDecimals('ANT')
-  const anjDecimals = useTokenDecimals('ANJ')
-  const usdcDecimals = useTokenDecimals('USDC')
+  const [amountOther, setAmountOther] = useState(bigNum(0))
+  const [editing, setEditing] = useState(null)
 
-  const tokenDecimals = useMemo(
-    () => (symbol === 'USDC' ? usdcDecimals : antDecimals),
-    [antDecimals, symbol, usdcDecimals]
+  const anjDecimals = useTokenDecimals('ANJ')
+  const otherDecimals = useTokenDecimals(otherSymbol)
+
+  // convertFromAnj is used as a toggle to execute a conversion to or from ANJ.
+  const [convertFromAnj, setConvertFromAnj] = useState(false)
+  const anjRate = useAnjRate(
+    otherSymbol,
+    (convertFromAnj ? amountAnj : amountOther).toString(),
+    convertFromAnj
   )
 
   // Reset the inputs anytime the selected token changes
   useEffect(() => {
-    setInputValueToken('')
+    setInputValueOther('')
     setInputValueAnj('')
     setAmountAnj(bigNum(0))
-    setAmountToken(bigNum(0))
-  }, [tokenToAnjRate])
+    setAmountOther(bigNum(0))
+  }, [otherSymbol])
+
+  // Calculate the ANJ amount from the other amount
+  useEffect(() => {
+    if (
+      anjDecimals === -1 ||
+      otherDecimals === -1 ||
+      anjRate.loading ||
+      convertFromAnj ||
+      editing === 'anj'
+    ) {
+      return
+    }
+
+    const amount = amountOther
+      .mul(bigNum(10).pow(anjDecimals - otherDecimals))
+      .mul(anjRate.rate)
+      .div(bigNum(10).pow(UNISWAP_PRECISION))
+
+    setAmountAnj(amount)
+    setInputValueAnj(formatUnits(amount, { digits: anjDecimals }))
+  }, [
+    amountOther,
+    anjDecimals,
+    anjRate,
+    convertFromAnj,
+    editing,
+    otherDecimals,
+  ])
+
+  // Calculate the other amount from the ANJ amount
+  useEffect(() => {
+    if (
+      anjDecimals === -1 ||
+      otherDecimals === -1 ||
+      anjRate.loading ||
+      !convertFromAnj ||
+      editing === 'other'
+    ) {
+      return
+    }
+
+    const amount = amountAnj
+      .div(bigNum(10).pow(anjDecimals - otherDecimals))
```

Comment:
I don't think this is likely to result in failing transactions, but we should prefer to do this truncation at the end (as an exact "reverse" of the `Other -> ANJ` path; going forward added 0s first, and going backwards, we should truncate decimal places last).

Otherwise, we'll be losing precision before we apply the exchange rate.